File: STRIVER/video/vid-5/pattern7/p7.cpp
//      *
//     * *
//    * * *
//   * * * *
//  * * * * *

#include <iostream>
using namespace std;

int main()
{
    int row = 5;

    for (int i = 0; i < row; i++)
    {
        // space
        for (int j = 0; j < row-i-1; j++)
        {
            cout<<" ";
        }
        // starts
        for (int j = 0; j < 2*i+1; j++)
        {
            cout<<"*";
        }
        // space
        for (int j = 0; j < row-i-1; j++)
        {
            cout<<" ";
        }
        cout<<endl;
    }
    return 0;
}

File: STRIVER/video/vid-5 (Patterns)/pattern7/p7.py
    #      *
    #     * *     
    #    * * *   
    #   * * * *  
    #  * * * * *  


def p7(row:int)->None:
    for i in range(row):
        # space
        for j in range(row-i-1):
            print(" ",end="")

        # stars
        for j in range(2*i+1):
            print("*",end="")

        # space
        for j in range(row-i-1):
            print(" ",end="")
        print("")

row = 5
p7(row)